### PR TITLE
rpm: Update virtualization packages

### PR DESCRIPTION
**Pull request by andreabolognani**
**What this PR does / why we need it**:

Updates virtualization packages. Specifically:

* libvirt 7.6.0-6 → 8.0.0-2
* QEMU 6.0.0-33 → 6.2.0-5

The `libvirt.org/go/libvirt` Go package is also updated to match.

**Release note**:

```release-note
This version of KubeVirt includes upgraded virtualization technology based on libvirt 8.0.0 and QEMU 6.2.0.
Each new release of libvirt and QEMU contains numerous improvements and bug fixes.
```

### CODE REVIEW

**Comment by andreabolognani** [comment]
At least some of the failures look legit.

One is caused by netcat being missing from the image - libvirt used to drag this in, but it no longer does. We could add it back manually, but maybe the test case could be reworked so that doing that is not necessary? Perhaps there are other uses of netcat outside of the testsuite.

I'm unable to reproduce one of the failures locally because

```
S [SKIPPING] [6.113 seconds]
[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system][sig-compute] VM Live Migration
tests/migration_test.go:97
  Starting a VirtualMachineInstance 
  tests/migration_test.go:360
    live migration cancelation
    tests/migration_test.go:2208
      should be able to cancel a migration
      vendor/github.com/onsi/ginkgo/extensions/table/table.go:92
        [sig-storage][storage-req][test_id:2732] with RWX block disk and virtctl [It]
        vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:41

        Skip test when Block storage is not present
```

I'm unclear on what I need to do here, advice would be appreciated :)

I'll look into the whole thing more tomorrow.

**Comment by alicefr** [comment]
@andreabolognani I'll have a try but generally if you succeed with the examples, then it is what we expect to work with the setup.


> At least some of the failures look legit.
> 
> One is caused by netcat being missing from the image - libvirt used to drag this in, but it no longer does. We could add it back manually, but maybe the test case could be reworked so that doing that is not necessary? Perhaps there are other uses of netcat outside of the testsuite.
> 
> I'm unable to reproduce one of the failures locally because
> 
> ```
> S [SKIPPING] [6.113 seconds]
> [rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system][sig-compute] VM Live Migration
> tests/migration_test.go:97
>   Starting a VirtualMachineInstance 
>   tests/migration_test.go:360
>     live migration cancelation
>     tests/migration_test.go:2208
>       should be able to cancel a migration
>       vendor/github.com/onsi/ginkgo/extensions/table/table.go:92
>         [sig-storage][storage-req][test_id:2732] with RWX block disk and virtctl [It]
>         vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:41
> 
>         Skip test when Block storage is not present
> ```
> 
> I'm unclear on what I need to do here, advice would be appreciated :)
> 
> I'll look into the whole thing more tomorrow.

You are missing the storage class that support block storage. In this case ceph, you need to set `export KUBEVIRT_STORAGE="rook-ceph-default"` before creating the cluster with `make cluster-up` then it is installed and you can check it with `kubectl get sc`

**Comment by alicefr** [comment]
Some of the missing packages are:
  - e2fsprogs-*: this should be fine because we use the tool in the fixed appliance, so no need in the container
  - dhcp-*: we don't need any network as we are not operating on VMs but only on disks
  - dnf-*: this is fine because we don't need to install or fetch the dependencies
  - findutlis: I'll readd this
  - gdisk: this I'm not sure
  - genisoimage: this either I'm not sure
  - squashfs-tools: same as for e2fsprogs
  - supermin : this is fine as we use a fixed appliance

I'll try to find out in which cases libguestfs uses gdisk and genisoimage

**Comment by andreabolognani** [comment]
> You are missing the storage class that support block storage. In this case ceph, you need to set `export KUBEVIRT_STORAGE="rook-ceph-default"` before creating the cluster with `make cluster-up` then it is installed and you can check it with `kubectl get sc`

@alicefr thanks, I've tried this and it worked as expected. Did you find this documented anywhere, or is it tribal knowledge? Either way, I couldn't reproduce the failure locally so I'm going to assume it was a fluke until proven otherwise.

> findutlis: I'll readd this
>
> I'll try to find out in which cases libguestfs uses gdisk and genisoimage
> 
> they are listed in the package list for the appliance so I guess they are used there.

So you think that findutils, gdisk and genisoimage should all be added back to the guestfs image? If they're used by libguestfs at runtime even when working with a pre-generated appliance, isn't it a packaging bug that they're not dragged in explicitly? We should discuss this with Rich.

For the time being, I believe I have fixed one of the test failures. More to come next week :)

**Comment by alicefr** [comment]
@andreabolognani I think we should only add findutils as standard tool but gdisk and genisoimage no. I don't think we have a good split for using a fixed appliance in centos 8.

About the env variable, yes I don't think that is documented :/. This script uses some of the variables in KubeVirt CI: https://github.com/kubevirt/kubevirt/blob/main/automation/test.sh

**Comment by andreabolognani** [comment]
Shouldn't be an issue since a couple of tests are still failing, but just to be on the safe side

/hold

**Comment by andreabolognani** [comment]
I'm still investigating the issue with booting the Alpine Linux 3.7 image using the RHEL 8.6 machine type, but in the meantime I've brought all images into the current decade :)

/hold cancel

**Comment by andreabolognani** [comment]
Most of the failures are probably caused by the ongoing [CI cluster upgrade](https://groups.google.com/g/kubevirt-dev/c/_NNMMTeCTpw), I assume.

@rmohr @vladikr can you please take a look? I think this should be ready to be merged now :)

**Comment by rmohr** [comment]
> Most of the failures are probably caused by the ongoing [CI cluster upgrade](https://groups.google.com/g/kubevirt-dev/c/_NNMMTeCTpw), I assume.

Probably. It looks weird.

/retest

File: WORKSPACE [hunk 1]
```diff
@@ -87,46 +87,41 @@ http_archive(
...
+        "https://download.cirros-cloud.net/0.5.2/cirros-0.5.2-aarch64-disk.img",
     ],
 )
 
 http_file(
     name = "microlivecd_image",
-    sha256 = "ae449ae8c0f73b1a7e2c394bc5385e7ab01d8fc000f5b074bc8b2aaabf931eac",
+    sha256 = "4f50fb32adb5355259101432da0235f8fc65843e3f324626243fcd59799a73c1",
     urls = [
-        "https://github.com/jean-edouard/microlivecd/releases/download/0.1/microlivecd_amd64.iso",
```
Comment (rmohr) [inline WORKSPACE:128]: This can finally be dropped when https://github.com/kubevirt/kubevirt/pull/6958. Hopefully the last bump :)

**Comment by rmohr** [comment]
/lgtm
/approve

If CI is happy, I am :) Thanks @andreabolognani 

One question: Since you are going through this procedure not the first time, what are the most rough edges where you would think that bazeldnf could support you more? I see for instance that there was a broken symlink added. Did you have to investigate that or would you expect that bazeldnf fails the build if the symlink specified is broken? Other things coming to your mind?

**Comment by andreabolognani** [comment]
> One question: Since you are going through this procedure not the first time, what are the most rough edges where you would think that bazeldnf could support you more? I see for instance that there was a broken symlink added. Did you have to investigate that or would you expect that bazeldnf fails the build if the symlink specified is broken? Other things coming to your mind?

The main points of friction for me are how long it takes for `make rpm-deps` to complete and how frequently I need to re-run it because of merge conflicts. I have a pretty beefy workstation and each run still takes several minutes, with the CPU cores being severely underutilized the entire time. More parallelism would probably help, but I can't see that happening without a significant rework of how bazeldnf works.

The broken symlinks, I noticed by pure chance. It would have been neat if bazeldnf could have figured that out on its own and reported it.

One thing that is not bazeldnf related but that took me way longer than I would have liked is figuring out the new values that I needed to use in `Expect()` calls. When the value you have in your test is not the one that's actually produced at runtime, you'll just get a timeout and produced output is not recorded anywhere AFAICT. The way I worked around this was to replace e.g.

```go
[]expect.Batcher{
    &expect.BSnd{S: "echo $var"},
    &expect.BExp{R: console.RetValue("foo")},
}
```

with

```go
[]expect.Batcher{
    &expect.BSnd{S: "echo $var"},
    &expect.BExp{R: console.PromptExpression},
    &expect.BSnd{S: "echo $var"},
    &expect.BExp{R: console.RetValue("foo")},
}
```

so that the last interaction would still time out, but in the meantime the first one would have succeeded and its output displayed by `make functest`. There's probably a better way to do it, but that's what I could come up with :)

File: WORKSPACE (hunk 1 above)
Comment (andreabolognani) [reply WORKSPACE:128]: That's excellent news! I was just thinking the other day that the Fedora 32 based image should also be replaced with something more modern, but as doing so in this case would have been significantly more involved than just changing a few URLs I decided to leave it for another day :)

**Comment by andreabolognani** [comment]
The network test is still failing

```
09:13:29: $ ping kubevirt.io -c 1 -w 5 -s 1452
09:13:29: PING kubevirt.io (185.199.111.153): 1452 data bytes
09:13:29: 
09:13:29: --- kubevirt.io ping statistics ---
09:13:29: 1 packets transmitted, 0 packets received, 100% packet loss
```

I have no idea what the problem could be. It works fine when I run it locally.

**Comment by rmohr** [comment]
> I have no idea what the problem could be. It works fine when I run it locally.

/hold

 to stop automatic retesting.

**Comment by rmohr** [comment]
> The main points of friction for me are how long it takes for `make rpm-deps` to complete and how frequently I need to re-run it because of merge conflicts. I have a pretty beefy workstation and each run still takes several minutes, with the CPU cores being severely underutilized the entire time. More parallelism would probably help, but I can't see that happening without a significant rework of how bazeldnf works.

Hm, I could try downloading and reading the XML files in parallel. I think we wait more than 90% of the whole time just on reading the XML files because the built in XML parser is really slow, but I wanted to avoid the need to have XML parsing shared libraries. The more channels/repos we add, the more time passes just on reading the XML files. Will check it out.

**Comment by andreabolognani** [comment]
I can reproduce at least some of the failures from `pull-kubevirt-check-tests-for-flakes` locally. Looking into them.

> Hm, I could try downloading and reading the XML files in parallel. I think we wait more than 90% of the whole time just on reading the XML files because the built in XML parser is really slow, but I wanted to avoid the need to have XML parsing shared libraries. The more channels/repos we add, the more time passes just on reading the XML files. Will check it out.

Without having actually profiled the tool, I would assume that a limiting factor is that we do repeated invocations of the tool to produce each rpmtree, each with its overhead in terms of startup and data loading, and they're all serialized. To minimize execution time, ideally you'd be able to do something like (simplified)

```
bazeldnf fetch
bazeldnf rpmtree \
         --name libvirt-devel_x86_64 --arch x86_64 $libvirtdevel_x86_64 \
         --name libvirt-devel_aarch64 --arch aarch64 $libvirtdevel_aarch64 \
         --name launcherbase_x86_64--arch x86_64 $launcherbase_x86_64 \
         --name launcherbase_aarch64 --arch aarch64 $launcherbase_aarch64 \
         ...
bazeldnf prune
```

so that a lot of the computation can happen as part of a single invocation, with data loaded from disk only once and each rpmtree being calculated at the same time as the others across multiple cores. Even if you had to handle each arch separately, I think the result might still be quite a bit faster.

Another idea, which is actually orthogonal to this one, would be to preprocess the data as part of the fetch step and write to disk a more efficient representation, such as a JSON file that only contains the bare minimum needed for the next steps.

**Comment by andreabolognani** [comment]
/retest

The network test still fails :(

**Comment by rmohr** [comment]
> The network test still fails :(

I think I see the issue: In the CI cluster we have a MTU of `1450`. 

I see that we try to send

```
ping kubevirt.io -c 1 -w 5 -s 1452
```


while we should probably send something like

```
ping kubevirt.io -c 1 -w 5 -s 1402
```

Since we anyway check if we can send full frames between VMIs (basically checking if inter-network config is correct), we can probably skip in the internet reach test the MTU check. We would always have to consider the cluster out-bound which is hard to know and I would prefer not setting it "manually".

**Comment by rmohr** [comment]
> Since we anyway check if we can send full frames between VMIs (basically checking if inter-network config is correct), we can probably skip in the internet reach test the MTU check. We would always have to consider the outer bound of the cluster which is hard to know and I would prefer not setting it "manually", since it does not really give additional test feedback.

Oh, while I think we should just skip this test, it is probably still worth finding out why we detected before this PR another MTU size ...

@qinqon any ideas?

**Comment by andreabolognani** [comment]
@rmohr I'll openly admit that networking is far from my strongest suit and your explanation went almost completely above my head :) How do I translate it into a patch that fixes the CI issues? Feel free to just make the change yourself and add it to the PR if that's more convenient than guiding me through it :)

**Comment by rmohr** [comment]
> @rmohr I'll openly admit that networking is far from my strongest suit and your explanation went almost completely above my head :) How do I translate it into a patch that fixes the CI issues? Feel free to just make the change yourself and add it to the PR if that's more convenient than guiding me through it :)

I think the smallest possible change would be to simply skip this check at https://github.com/kubevirt/kubevirt/blob/main/tests/network/vmi_networking.go#L188 when `vmiRef` is nil (which seems to indicate the internet reach scenario).

I would also be open to move the internet reach test out of this table and remove all MTU detection traces from it.

**Comment by andreabolognani** [comment]
> I think the smallest possible change would be to simply skip this check at https://github.com/kubevirt/kubevirt/blob/main/tests/network/vmi_networking.go#L188 when `vmiRef` is nil (which seems to indicate the internet reach scenario).
> 
> I would also be open to move the internet reach test out of this table and remove all MTU detection traces from it.

@rmohr I went for the latter as it seemed cleaner. Hopefully I didn't mess that up too badly :)

**Comment by andreabolognani** [comment]
@rmohr All tests passed now \o/ If the changes look good, can you please approve? Thanks!

**Comment by rmohr** [comment]
> > Since we anyway check if we can send full frames between VMIs (basically checking if inter-network config is correct), we can probably skip in the internet reach test the MTU check. We would always have to consider the outer bound of the cluster which is hard to know and I would prefer not setting it "manually", since it does not really give additional test feedback.
> 
> Oh, while I think we should just skip this test, it is probably still worth finding out why we detected before this PR another MTU size ...
> 
> @qinqon any ideas?

@qinqon @AlonaKaplan @maiqueb could you quickly check what could be different here? I am pretty sure it is ok what we did to work around this, but I am not entirely sure why we seem to detect now different MTU sizes.

**Comment by maiqueb** [comment]
> > > Since we anyway check if we can send full frames between VMIs (basically checking if inter-network config is correct), we can probably skip in the internet reach test the MTU check. We would always have to consider the outer bound of the cluster which is hard to know and I would prefer not setting it "manually", since it does not really give additional test feedback.
> > 
> > 
> > Oh, while I think we should just skip this test, it is probably still worth finding out why we detected before this PR another MTU size ...
> > @qinqon any ideas?
> 
> @qinqon @AlonaKaplan @maiqueb could you quickly check what could be different here? I am pretty sure it is ok what we did to work around this, but I am not entirely sure why we seem to detect now different MTU sizes.

I think the goal is to check that we correctly configure the MTU for all involved network elements - meaning that you can communicate e2e using different MTUs.

IMO, that makes sense in intra-cluster communication, where we own the network elements. IIUC, the issue was with reaching the internet - where we do not own the network segments - and as such cannot assume we can communicate with different MTUs. Thus, I am OK with your proposed fix @andreabolognani / @rmohr .

**Comment by rmohr** [comment]
> IMO, that makes sense in intra-cluster communication, where we own the network elements. IIUC, the issue was with reaching the internet - where we do not own the network segments - and as such cannot assume we can communicate with different MTUs. Thus, I am OK with your proposed fix @andreabolognani / @rmohr .

Thanks for the confirmation!

**Comment by AlonaKaplan** [comment]
/hold
It indeed makes sense to skip the MTU test when pinging the internet.
But if the reason for the failure is that the destination or some outer switch drops the packet, we should've see it before this PR. Upgrading libvirt/qemu doesn't effect the outer switch behaviour.
I think it worth spending some more time to investigate why upgrading the libvirt/qemu caused the test to start failing.

**Comment by rmohr** [comment]
> It indeed makes sense to skip the MTU test when pinging the internet.
> But if the reason for the failure is that the destination or some outer switch drops the packet, we should've see it before this PR. Upgrading libvirt/qemu doesn't effect the outer switch behaviour.
> I think it worth spending some more time to investigate why upgrading the libvirt/qemu caused the test to start failing.

Good point. I wonder what was discovered before this PR. Because the DIND interface in the CI clusters have 1450 MTU size and something must have changed, which leads to bigger frames in the kubevirtci cluster. It seems like it has to be in the cluster, otherwise the VM-to-VM communication would fail on the tests too ...

**Comment by AlonaKaplan** [comment]
> > It indeed makes sense to skip the MTU test when pinging the internet.
> > But if the reason for the failure is that the destination or some outer switch drops the packet, we should've see it before this PR. Upgrading libvirt/qemu doesn't effect the outer switch behaviour.
> > I think it worth spending some more time to investigate why upgrading the libvirt/qemu caused the test to start failing.
> 
> Good point. I wonder what was discovered before this PR. Because the DIND interface in the CI clusters have 1450 MTU size and something must have changed, which leads to bigger frames in the kubevirtci cluster. It seems like it has to be in the cluster, otherwise the VM-to-VM communication would fail on the tests too ...

The ping doesn't have `-M do`, it means fragmentation is allowed. So theoretically, even if the destination MTU is lower that the packet size, the ping should succeed. But I think it depends on the switch, so maybe some outer switch drops the packet.
Maybe the change is that libvirt/qemu adds some extra overhead to the header. I will try to debug it tomorrow.

**Comment by rmohr** [comment]
> I will try to debug it tomorrow.

Thanks! :+1:

**Comment by andreabolognani** [comment]
> I will try to debug it tomorrow.

TIA @AlonaKaplan :)

**Comment by AlonaKaplan** [comment]
> > > I will try to debug it tomorrow.

@andreabolognani  @rmohr @maiqueb 
We have 2 problems in our original tests -
1. Testing MTU without setting `-M do` on the `ping` is useless since fragmentation is allowed and ping to any packet size, even bigger than the allowed MTU, will work.
2. Testing MTU by pinging to external address is wrong since we don't control the MTU of the outside switches.

**The reason the original test works** (before this PR) -  is since **`-M do` (prohibit fragmentation) wasn't set** on the ping, so even though one of the hops in the way from the VM to `kubevirt.io` had smaller MTU than the packet size, the packet was fragmented and the ping succeeded.

**The reason the test stopped working on this PR** - **cirros upgrade**.
cirros doesn't support `-M` option, that's why the original test doesn't have it (and doesn't really test the MTU).

I guess the **new cirros** image has some bug/behaviour change I don't understand, that causes the fragmentation in our case not to be done (if the cirros image is switched to fedora, [the test works even without the `-M do`](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7348/pull-kubevirt-e2e-k8s-1.21-sig-network/1502238814117040128#1:build-log.txt%3A5763)).

Since it anyway doesn't make sense to test MTU with ping to the internet and the test has to forbid fragmentation, it is not very interesting to understand why fragmentation worked in the old image and stopped working in the new (the new image seems to support local fragmentation).

How I suggest to fix the PR -
- Removing the MTU internet test is already done, and I think it is correct.
- [Use fedora image with `-M do` specified on the `ping` instead of cirros](https://github.com/kubevirt/kubevirt/pull/7348/commits/cb1fee9ef7f75a2724940251934e6724835f4070#diff-6fdd74ae3ebff086dcf12ed84c781a5ae997ea4efae4763ec864e57b6afc1807R270).
- [Test that pinging to `packetSize +1` fails](https://github.com/kubevirt/kubevirt/pull/7348/commits/cb1fee9ef7f75a2724940251934e6724835f4070#diff-6fdd74ae3ebff086dcf12ed84c781a5ae997ea4efae4763ec864e57b6afc1807R272).

References to my tests-
[Original code](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7348/pull-kubevirt-e2e-k8s-1.21-sig-network/1502238814117040128)
[This PR included](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7352/pull-kubevirt-e2e-k8s-1.21-sig-network/1502239052873601024)
[This PR includes but cirros update reverted](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7356/pull-kubevirt-e2e-k8s-1.21-sig-network/1502239084070834176)

**Comment by andreabolognani** [comment]
I've rebased the PR on top of main and picked up a more recent build of edk2 in the process.

@AlonaKaplan thank you for your analysis! What you describe mostly makes sense to me, although as I have mentioned before networking is very far from being my area of expertise O:-)

Since the changes that I have made to the networking tests as part of this PR are okay, do you think it would be possible to merge this as-is and make the other changes as part of a follow-up PR? I could try preparing such a follow-up PR based on your sample code, or you could polish up what you already have and submit it yourself - whatever works best for you.

**Comment by AlonaKaplan** [comment]
> Since the changes that I have made to the networking tests as part of this PR are okay, do you think it would be possible to merge this as-is and make the other changes as part of a follow-up PR? I could try preparing such a follow-up PR based on your sample code, or you could polish up what you already have and submit it yourself - whatever works best for you.

We can merge it as is. The network team will work on the follow-up.

/hold cancel

**Comment by andreabolognani** [comment]
> We can merge it as is. The network team will work on the follow-up.

@AlonaKaplan sounds awesome, thanks!

@rmohr can you please re-lgtm this?

**Comment by andreabolognani** [comment]
Rebased to address a merge conflict.

@rmohr can you please re-lgtm? Thanks!

**Comment by rmohr** [comment]
/lgtm

Thanks!

**Comment by rmohr** [comment]
@andreabolognani sorry for asking you, but could you bump the RPM dependencies again? We did not yet mirror them and some of them are gone already.

**Comment by andreabolognani** [comment]
> @andreabolognani sorry for asking you, but could you bump the RPM dependencies again? We did not yet mirror them and some of them are gone already (parts of the cluster have it still cached, therefore not all lanes are failing).

Weird, I ran `make rpm-deps` and it didn't generate any changes. To be fair, I haven't compile-tested the branch either because I assumed things were good based on that... Let me give it a closer look this time.

**Comment by rmohr** [comment]
@andreabolognani let's see if it is temporary.

**Comment by andreabolognani** [comment]
@rmohr I tried updating all packages again but it didn't pick up any additional changes.

When I tried deploying locally the first time around, I got a bunch of errors like

```
Error pushing image to localhost:49160/kubevirt/example-cloudinit-hook-sidecar:devel: unable to push image to localhost:49160/kubevirt/example-cloudinit-hook-sidecar:devel: Get "https://localhost:49160/v2/": tls: first record does not look like a TLS handshake; Get "http://localhost:49160/v2/": net/http: HTTP/1.x transport connection broken: malformed HTTP response "SSH-2.0-OpenSSH_8.0"
```

After I cleaned up everything and tried again, they went away and KubeVirt seems to be working fine based on a very quick test.

Hopefully the CI cluster is not affected by whatever local state shenanigans were happening on my machine :)

**Comment by rmohr** [comment]
/lgtm
/retest

seems like the error went away :)

**Comment by andreabolognani** [comment]
Rebased on top of main and picked up a few CentOS Stream updates in the process.

I see that PRs have been merged as late as yesterday, so I'm hoping that the CI issues were either temporary or have been addressed separately since last week. I'd really like to get this in :)

**Comment by rmohr** [comment]
/retest

Can be that we have to skip the failing test on ipv6. Will ensure that we have this finally ready today.

**Comment by rmohr** [comment]
@andreabolognani, @AlonaKaplan  added a skip to the internet test if it is run on ipv6 only clusters.

Regarding vgpu and sriov, we still have issues which started over the weekend on them ... :/

**Comment by andreabolognani** [comment]
> added a skip to the internet test if it is run on ipv6 only clusters.

Thanks!

> Regarding vgpu and sriov, we still have issues which started over the weekend on them ... :/

Fingers crossed then O:-)

**Comment by EdDev** [comment]
@AlonaKaplan , can you please have a look?
The `sig-network` label was raised, I am not sure if anyone from the sig had a chance to see these changes.

**Comment by rmohr** [comment]
@AlonaKaplan done. Thanks.

**Comment by AlonaKaplan** [comment]
The ipv6 test is still not working since we don't have DHCP6 so the VM is not familiar with the DNS IP.
The test should be skipped if the cluster doesn't support IPv4.
Sorry for the noise.

**Comment by AlonaKaplan** [comment]
> > The ipv6 test is still not working since we don't have DHCP6 so the VM is not familiar with the DNS IP.
> > The test should be skipped if the cluster doesn't support IPv4.
> > Sorry for the noise.
> 
> @AlonaKaplan does the diff below look reasonable to you? I'm preparing a rebase and I can just squash it in.
> 
> ```diff
> diff --git a/tests/network/vmi_networking.go b/tests/network/vmi_networking.go
> index ebac17309..1b0f30e25 100644
> --- a/tests/network/vmi_networking.go
> +++ b/tests/network/vmi_networking.go
> @@ -300,12 +300,13 @@ var _ = SIGDescribe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:c
>  
>         Context("VirtualMachineInstance with default settings", func() {
>                 It("[test_id:1542]should be able to reach the internet", func() {
> +                       libnet.SkipWhenClusterNotSupportIpv4(virtClient)
>                         outboundVMI := libvmi.NewCirros(
>                                 libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
>                                 libvmi.WithNetwork(v1.DefaultPodNetwork()),
>                         )
>                         outboundVMI = runVMI(outboundVMI)
> -                       Expect(libnet.WithIPv6(console.LoginToCirros)(outboundVMI)).To(Succeed())
> +                       tests.WaitUntilVMIReady(outboundVMI, console.LoginToCirros)
>  
>                         By("checking the VirtualMachineInstance can fetch via HTTP")
>                         err := console.SafeExpectBatch(outboundVMI, []expect.Batcher{
> ```

Yes. And the VM should have bridge binding as the original test had.

```
outboundVMI := libvmi.NewCirros()
```

Sorry again for my previous misleading request.

**Comment by andreabolognani** [comment]
> Sorry again for my previous misleading request.

No worries, thank you for looking into it and helping out :)

**Pull request merged**